rfcs/0051-emit-on-close-stream-executor.md

```diff
@@ -0,0 +1,143 @@
+---
+feature: Stream Executor with Emit on Close Semantics
+authors:
+  - "st1page"
+start_date: "2023/02/14"
+---
+
+# Stream Executor with Emit on Close Semantics 
+
+## Summary
+
+Introduce a new component Sort Buffer used in the streaming executor and give a general method to implement EMIT ON CLOSE stream executor. Sort Buffer can support materialize and persistent the changes stream and drain the "stable" records with the input watermark. Because the input watermarks are always monotonically increasing, the output records are ordered by the watermark column and append-only.
+
+***Note: In this RFC, we will focus on the EMIT ON CLOSE query***
+
+## Background and Definitions 
+Let's look back on the current designs and RFCs. 
+
+The [RFC: The Semantics of EMIT ON WINDOW CLOSE](https://github.com/risingwavelabs/rfcs/pull/30) has given a strict definition of the semantics of **EMIT ON WINDOW CLOSE**. In short, the streaming job with the "EMIT ON CLOSE" modifier emits the result when the result is complete, and no sooner, no later. In other words, the "EMIT ON CLOSE" streaming query gives an append-only result as soon as possible.
```

Comment:
`EMIT ON WINDOW CLOSE` and `EMIT ON COMPLETE` are different things, the former will ensure that a window is closed, and the latter only ensure that the result is append-only.